### PR TITLE
:sparkles: Collapse time in one day

### PR DESCRIPTION
**Pull request by sofokli1**
I tried to keep the logic as simple as possible

### CODE REVIEW

File: teferi/Extensions/DateExtensions.swift [hunk 1]
```diff
@@ -108,9 +108,12 @@ extension Date
         return calendar.date(from: currentComponents)!
     }
     
-    func timeIntervalBasedOnWeekDaySince(_ date: Date) -> TimeInterval
+    func timeIntervalForSameDaySince(_ date: Date) -> TimeInterval
```
Comment (Odrakir) [inline teferi/Extensions/DateExtensions.swift]: How about `timeIntervalIgnoringDate`. I don't want to be picky, just trying to find the best name so we remember what this does in the future.

File: teferi/Services/Implementations/Persistency/DefaultSmartGuessService.swift [hunk 2]
```diff
@@ -166,7 +166,7 @@ class DefaultSmartGuessService : SmartGuessService
             let smartGuessTimestamp = smartGuess.location.timestamp
             let locationTimestamp = location.timestamp
             
-            return abs(smartGuessTimestamp.timeIntervalBasedOnWeekDaySince(locationTimestamp)) <= self.timeThreshold
+            return abs(smartGuessTimestamp.timeIntervalForSameDaySince(locationTimestamp)) <= self.timeThreshold
```
Comment (Odrakir) [inline teferi/Services/Implementations/Persistency/DefaultSmartGuessService.swift]: The name of this method then, shouldn't be `isWithinTimeThresholdInNearByWeekDay` any more. How about, following my previous comment: `isWithinTimeThresholdIgnoringDate`?

File: teferi/Extensions/DateExtensions.swift (hunk 1 above)
Comment (amulware) [inline teferi/Extensions/DateExtensions.swift]: Is it worth including something like 'absolute' in the name, to make it clear that this does not return negative values?

File: teferi/Extensions/DateExtensions.swift [hunk 3]
```diff
@@ -108,9 +108,12 @@ extension Date
...
     {
-        let dayDifference = dayOfWeek - date.dayOfWeek
-        return timeIntervalSince(date.convert(calendarUnits: [ .year, .month, .day], sameAs: self).add(days: -dayDifference))
+        let differenceIfPreviousDay = abs(self.ignoreDateComponents().timeIntervalSince(date.ignoreDateComponents().add(days: -1)))
+        let differenceIfSameDay = abs(self.ignoreDateComponents().timeIntervalSince(date.ignoreDateComponents()))
+        let differenceIfNextDay = abs(self.ignoreDateComponents().timeIntervalSince(date.ignoreDateComponents().add(days: 1)))
```
Comment (amulware) [inline teferi/Extensions/DateExtensions.swift]: Might be nicer to cache the two times-without-date for readability, and performance?

**Pull request merged**